mailpoet/tests/integration/Automation/Engine/Storage/WorkflowStatisticsStorageTest.php

```diff
@@ -0,0 +1,192 @@
+    while($i < $expectedExited) {
+      $this->createRun($workflow, WorkflowRun::STATUS_FAILED);
+      $i++;
+    }
+
+    $statistics = $this->testee->getWorkflowStats($workflow->getId(), $versionId);
+    $this->assertEquals($expectedInProgress, $statistics->getInProgress());
+    $this->assertEquals($expectedTotal, $statistics->getEntered());
+    $this->assertEquals($expectedExited, $statistics->getExited());
+    $this->assertEquals((bool)$expectedTotal, $statistics->hasValues());
+    $this->assertEquals([
+      'workflow_id' => $workflow->getId(),
+      'has_values' => (bool) $expectedTotal,
+      'totals' => [
+        'entered' => $expectedTotal,
+        'in_progress' => $expectedInProgress,
+        'exited' => $expectedExited,
+      ],
+    ], $statistics->toArray());
+  }
+
+  public function dataForTestItCalculatesTotalsCorrectly() {
+    return [
+      'zero' => [
+        1, 0,0,0, null
+      ],
+      'two-one-one' => [
+        0, 2,1,1, null
+      ],
+      'two-two-zero' => [
+        2, 2,2,0, null
+      ],
+      'two-zero-two' => [
+        1, 2,0,2, null
+      ],
+    ];
+  }
+
+  public function testItSeperatesWorkflowRunsCorrectly() {
+    $workflow1 = $this->workflowStorage->getWorkflow($this->workflows[0]);
+    assert($workflow1 instanceof Workflow);
+    $workflow2 = $this->workflowStorage->getWorkflow($this->workflows[1]);
+    assert($workflow2 instanceof Workflow);
+    $workflow3 = $this->workflowStorage->getWorkflow($this->workflows[2]);
+    assert($workflow3 instanceof Workflow);
+
+    $this->createRun($workflow1, WorkflowRun::STATUS_RUNNING);
+    $this->createRun($workflow1, WorkflowRun::STATUS_COMPLETE);
+
+    $this->createRun($workflow2, WorkflowRun::STATUS_RUNNING);
+    $this->createRun($workflow2, WorkflowRun::STATUS_RUNNING);
+    $this->createRun($workflow2, WorkflowRun::STATUS_CANCELLED);
+    $this->createRun($workflow2, WorkflowRun::STATUS_CANCELLED);
+
+    $this->createRun($workflow3, WorkflowRun::STATUS_RUNNING);
+    $this->createRun($workflow3, WorkflowRun::STATUS_RUNNING);
+    $this->createRun($workflow3, WorkflowRun::STATUS_RUNNING);
+    $this->createRun($workflow3, WorkflowRun::STATUS_FAILED);
+    $this->createRun($workflow3, WorkflowRun::STATUS_FAILED);
+    $this->createRun($workflow3, WorkflowRun::STATUS_FAILED);
+
+    $statistics1 = $this->testee->getWorkflowStats($workflow1->getId(), $workflow1->getVersionId());
+    $this->assertEquals(1, $statistics1->getInProgress());
+    $this->assertEquals(1, $statistics1->getExited());
+
+    $statistics2 = $this->testee->getWorkflowStats($workflow2->getId(), $workflow2->getVersionId());
+    $this->assertEquals(2, $statistics2->getInProgress());
+    $this->assertEquals(2, $statistics2->getExited());
+
+    $statistics3 = $this->testee->getWorkflowStats($workflow3->getId(), $workflow3->getVersionId());
+    $this->assertEquals(3, $statistics3->getInProgress());
+    $this->assertEquals(3, $statistics3->getExited());
```

Comment:
With `testItSeparatesWorkflowRunsCorrectly` I just want to check whether the queries correctly account for the `workflow_id`. This is why I have different values per Workflow. If `getExited()` would change `testItCalculatesTotalsCorrectlyForSingleWorkflow` fails, which accounts for that. 

Thinking about it, I think on the contrary the `testItSeparatesWorkflowRunsCorrectly` test should be even more simple to avoid it failing in a scenario you describe, to improve the debugging experience. What do you think about 6fca21650

Reply:
Ah, I see. I thought the separation was referring to run status as well. I like your change!